### ISSUE TITLE
ELI5 Hosting?

### ISSUE THREAD

**Issue by poltergeisha** [Kruptein/PlanarAlly#72]
I'm really interested in trying this out (especially in the wake of the recent Roll20 dumpster fire), but I have not a clue how to host this. Can I use XAMPP? Or should I use something else?  It seems like the server needs to be hosted on some platform to work.  I have a very basic background in computer science so if you can give me a place to start it would go a long way.

**Comment by Kruptein** [comment]
Hey, the software itself is a webserver, so no XAMPP or alike is needed.

Have you seen the installation information in [the docs](https://planarally.readthedocs.io/en/latest/getting_started.html#manual-installation) ?

You'll need to install the dependencies and run the planarserver.py script

**Comment by schemen** [comment]
I created https://github.com/Kruptein/PlanarAlly/pull/73 for a docker based hosting, this should make it easier if you know your way around docker.

**Comment by poltergeisha** [comment]
I downloaded the windows zip and was trying to run the .exe with no success. @Kruptein

**Comment by Kruptein** [comment]
Can you explain what is not working ?
What happens if you run the exe?

**Comment by poltergeisha** [comment]
![image](https://user-images.githubusercontent.com/8014948/46171714-f3f25d80-c266-11e8-84dd-4319f960f55a.png)

**Comment by Kruptein** [comment]
Oh well that means everything is fine! You need to open a browser and visit
'http://localhost:8000'

I've had some other people with the same confusion, I'll try to integrate
it in the documentation.

**Comment by poltergeisha** [comment]
Ah, so what URL would I give my players so they could connect?

**Comment by poltergeisha** [comment]
Ah nvm, I found it I think.

**Comment by poltergeisha** [comment]
So I'm going to have to figure out how to host this not on localhost for my players to be able to connect from across teh waves, right?

**Comment by Kruptein** [comment]
I wrote a bit of information regarding your "problem" on reddit as a reaction to someone with a similar issue.  https://www.reddit.com/r/DnD/comments/9j3vzp/tool_planarally_v09_a_selfhosted_virtual_tabletop/e6q76xa/

I'll try to improve the user documentation in the coming days/weeks to explain this process as well.

**Comment by poltergeisha** [comment]
For anyone else reading this who's looking for a way to avoid the port forwarding thing, you should try [ngrok](https://ngrok.com)

**Comment by Kruptein** [comment]
Hey, it's `/invite/<code>` :) for example `
http://localhost.com:8000/invite/g63d74fh8yy7hh`

**Comment by Kruptein** [comment]
For some reason I only got half your message earlier in my mail.

The 401 Unauthorized is normal, it means the user is not logged in yet.
First let your user(s) go to the homepage and let them make an account.  Then let them go to the invite link.

I know it's not ideal, it's on my list of todos to automaticlaly reroute them to the login screen if they are not yet logged in, in the future.
